### PR TITLE
Add Distribution Release bundle Task

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/artifactory-azure-devops-extension#testing) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: tests/testUtils.ts [hunk 1]
```diff
@@ -81,6 +92,16 @@ export function initTests(): void {
 }
 
 export function runTask(testMain: string, variables: any, inputs: any): void {
```
Comment (eyalbe4) [inline tests/testUtils.ts:95]: runTask --> runArtifactoryTask

File: tests/testUtils.ts [hunk 2]
```diff
@@ -81,6 +92,16 @@ export function initTests(): void {
...
+    setServiceConnectionCredentials(artifactoryUrl);
+    runTaskForService(testMain, variables, inputs);
+}
+
+export function runDistTask(testMain: string, variables: any, inputs: any): void {
```
Comment (eyalbe4) [inline tests/testUtils.ts:100]: runDistTask --> runDistributionTask

File: tests/testUtils.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ const artifactoryUrl: string = process.env.ADO_ARTIFACTORY_URL || '';
 const artifactoryUsername: string = process.env.ADO_ARTIFACTORY_USERNAME || '';
 const artifactoryPassword: string = process.env.ADO_ARTIFACTORY_PASSWORD || '';
 const artifactoryAccessToken: string = process.env.ADO_ARTIFACTORY_ACCESS_TOKEN || '';
+const distributionUrl: string = process.env.ADO_DISTRIBUTION_URL || '';
 const artifactoryDockerDomain: string = process.env.ADO_ARTIFACTORY_DOCKER_DOMAIN || '';
 const artifactoryDockerRepo: string = process.env.ADO_ARTIFACTORY_DOCKER_REPO || '';
 const skipTests: string[] = process.env.ADO_ARTIFACTORY_SKIP_TESTS ? process.env.ADO_ARTIFACTORY_SKIP_TESTS.split(',') : [];
```
Comment (eyalbe4) [inline tests/testUtils.ts:20]: ADO_ARTIFACTORY_SKIP_TESTS --> ADO_SKIP_TESTS

File: README.md [hunk 4]
```diff
@@ -52,7 +57,7 @@ Note: If you are running tests via your IDE, make sure you are registering tests
 
 ### Skipping Tests
 In order to skip tests, set the ADO_ARTIFACTORY_SKIP_TESTS environment variable with the tests you wish to skip, separated by commas.
```
Comment (eyalbe4) [inline README.md:59]: ADO_ARTIFACTORY_SKIP_TESTS --> ADO_SKIP_TESTS

File: vss-extension.json [hunk 5]
```diff
@@ -136,7 +138,91 @@
...
+                    }
+                ],
+                "authenticationSchemes": [
+                    {
+                        "type": "ms.vss-endpoint.endpoint-auth-scheme-basic",
+                        "inputDescriptors": [
+                            {
+                                "id": "username",
+                                "name": "Username",
+                                "description": "Distribution username with deploy permissions.",
```
Comment (eyalbe4) [inline vss-extension.json:183]: Are you sure the "with deploy permissions" part is relevant for Distribution?

File: vss-extension.json [hunk 6]
```diff
@@ -136,7 +138,91 @@
...
+                                "inputMode": "textbox",
+                                "validation": {
+                                    "isRequired": true,
+                                    "dataType": "string"
+                                }
+                            },
+                            {
+                                "id": "password",
+                                "name": "Password",
+                                "description": "Password or API key of an Distribution user with deploy permissions.",
```
Comment (eyalbe4) [inline vss-extension.json:193]: Are you sure the "API key" and "with deploy permissions" parts are relevant for Distribution?

File: vss-extension.json [hunk 7]
```diff
@@ -136,7 +138,91 @@
...
+                },
+                "icon": "images/distribution.png",
+                "dataSources": [
+                    {
+                        "name": "TestConnection",
+                        "endpointUrl": "{{endpoint.url}}/api/v1/system/ping",
+                        "resultSelector": "jsonpath:$.values[*]"
+                    },
+                    {
+                        "name": "BuildNames",
```
Comment (eyalbe4) [inline vss-extension.json]: Is this field required for the Distribution service connection? Same goes for the below "BuildNumbers".
These fields may also be redundant in the Artifactory service connection.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: vss-extension.json (hunk 7 above)
Comment (RobiNino) [reply vss-extension.json]: Already deleted from distribution, pretty sure they are needed in Artifactory for the artifact source

**Pull request merged**